synapse/replication/tcp/client.py

```diff
@@ -207,6 +207,11 @@ async def on_rdata(
             # we don't need to optimise this for multiple rows.
             for row in rows:
                 if row.type != EventsStreamEventRow.TypeId:
+                    # The row's data is an `EventsStreamCurrentStateRow`.
+                    # When finishing up a partial state join, the current state of a
+                    # room will change without any new events being persisted, so we
+                    # must poke the replication callbacks ourselves.
+                    self.notifier.notify_replication()
```

Comment:
This looks like a trick, shouldn't we do that a bit further down in `elif stream_name == UnPartialStatedRoomStream.NAME:` case ?

Reply:
The room un-partial stating process looks like:

https://github.com/matrix-org/synapse/blob/cdea7c11d082e73606bea5d0462f7971e90d836c/synapse/handlers/federation.py#L1806-L1820

`update_current_state` sends the `EventsStreamCurrentStateRow` over replication, whereas `clear_partial_state_room` sends the `UnPartialStatedRoomStreamRow`.

My thinking is that I

Reply:
I've reworded the comment to place less emphasis on faster joins.